src/openms/source/METADATA/ExperimentalDesign.cpp

```diff
@@ -599,8 +599,55 @@ namespace OpenMS
       std::set< std::tuple< std::string, unsigned > > path_label_set;
       std::map< std::tuple< unsigned, unsigned >, std::set< unsigned > > fractiongroup_label_to_sample;
 
+      if (msfile_section_[0].fraction != 1)
+      {
+        throw Exception::InvalidValue(
+            __FILE__,
+            __LINE__,
+            OPENMS_PRETTY_FUNCTION, "Fractions do not start with 1.", String(msfile_section_[0].fraction));
+      }
+      if (msfile_section_[0].fraction_group != 1)
+      {
+        throw Exception::InvalidValue(
+            __FILE__,
+            __LINE__,
+            OPENMS_PRETTY_FUNCTION, "Fraction groups do not start with 1.", String(msfile_section_[0].fraction_group));
+      }
+
+      Size last_fraction = 1;
+      Size last_fraction_group = 0;
       for (const MSFileSectionEntry& row : msfile_section_)
```

Comment:
add comment above for loop:
// ensure that fraction group and fraction identifier are consecutive numbers

Reply:
isnt that obvious from the exception?

Reply:
and the loop does not only do that